src/hash.cr

```diff
@@ -572,9 +572,12 @@ class Hash(K, V)
     # while moving non-deleted entries to the beginning (compaction).
     new_entry_index = 0
     each_entry_with_index do |entry, entry_index|
+      entry_hash = entry.hash
```

Comment:
Wouldn't this be better placed in the `else` branch?

Reply:
Maybe, I'm not sure what's the difference.

Reply:
In this way you'd avoid one unnecessary call.
I guess that was @straight-shoota's point.

Reply:
Sounds good!

(just note that `entry.hash` is just accessing an instance variable from a struct and that's like super cheap, plus I'm almost sure that LLVM will figure out that `entry_hash` is only reassigned in the `if` branch and so the right thing)

Reply:
Performance shouldn't be an issue. But it's easier to understand the code. This assignment is only relevant in the else branch. So it should only be there.